servers/graphql-kotlin-server/src/main/kotlin/com/expediagroup/graphql/server/execution/GraphQLContextFactory.kt

```diff
@@ -16,20 +16,10 @@
 
 package com.expediagroup.graphql.server.execution
 
-import com.expediagroup.graphql.generator.execution.GraphQLContext
-
 /**
  * Factory that generates GraphQL context.
  */
-interface GraphQLContextFactory<out Context : GraphQLContext, Request> {
-
-    /**
-     * Generate GraphQL context based on the incoming request and the corresponding response.
-     * If no context should be generated and used in the request, return null.
-     */
-    @Deprecated(message = "The generic context object is deprecated in favor of the context map", replaceWith = ReplaceWith("generateContextMap(request)"))
-    suspend fun generateContext(request: Request): Context? = null
-
+interface GraphQLContextFactory<Request> {
```

Comment:
Maybe we can default to `GraphQLContext.of()` to be extra clear what the object here is

Reply:
Actually, we could make a MUCH more significant change and instead start using `GraphQLContext` as the type everywhere instead of `Map<*, Any>?`. 

We end up mapping back to a builder anyway in `GraphQLRequest.toExecutionInput()` so if we let everyone just call the builder themselves to modify that might be eaiser

Reply:
That would require everyone to migrate though, and almost certainly break existing code.Tthis is at least a breaking change but will possibly only require removal as someone could have migrated to the new `Map` methods

Reply:
i think for simplicity it was decided to just return a `Map` and let the `GraphQLContextFactory` build the `GraphQLContext` using the map. 
i would keep it like that for simplicity and to not add another breaking change bc of the potential change of the return type

Reply:
since it is a breaking change anyway -> lets approach it from a new user perspective, what would be a better user experience?

Reply:
So `DataFetchingEnvironment.getGraphQlContext` returns `GraphQLContext` not `Map<>` so I think it would sense that the factory sets an object and that is the one you get back out.

Reply:
applied the updates to return a `GraphQLContext` object instead of a Map